tests/opera/parser/yaml/test_constructor.py

```diff
@@ -0,0 +1,140 @@
+            ("0x90abc", 592572), ("0xAaBbFdE", 179027934),
+        ],
+    )
+    def test_construct_int(self, value, parse):
+        mark = Mark(None, None, 3, 7, None, None)
+        node = ScalarNode(None, value, start_mark=mark)
+        res = Constructor("int").construct_yaml_int(node)
+
+        assert res.value == parse
+        assert res.loc.line == 4
+        assert res.loc.column == 8
+        assert res.loc.stream_name == "int"
+
+    @pytest.mark.parametrize(
+        "value,parse", [
+            ("+.inf", math.inf), ("-.inf", -math.inf), (".inf", math.inf),
+            ("+.Inf", math.inf), ("-.Inf", -math.inf), (".Inf", math.inf),
+            ("+.INF", math.inf), ("-.INF", -math.inf), (".INF", math.inf),
+            ("+.987", .987), ("-.765", -.765), (".0987", .0987),
+            ("+.6e-3", .6e-3), ("-.5E+2", -.5e2), (".4E32", .4e32),
+            ("+1.3", 1.3), ("-2.4", -2.4), ("3.5", 3.5), ("+1.3E-3", 1.3e-3),
+            ("-2.42e+5", -2.42e5), ("3.5e7", 3.5e7), ("+13E-3", 13e-3),
+            ("-2e+5", -2e5), ("3E7", 3e7),
+        ],
+    )
+    def test_construct_float_no_nan(self, value, parse):
+        mark = Mark(None, None, 2, 8, None, None)
+        node = ScalarNode(None, value, start_mark=mark)
+        res = Constructor("float").construct_yaml_float(node)
+
+        assert res.value == parse
+        assert res.loc.line == 3
+        assert res.loc.column == 9
+        assert res.loc.stream_name == "float"
+
+    @pytest.mark.parametrize("value", [".nan", ".NaN", ".NAN"])
+    def test_construct_float_nan_only(self, value):
+        mark = Mark(None, None, 7, 1, None, None)
+        node = ScalarNode(None, value, start_mark=mark)
+        res = Constructor("float").construct_yaml_float(node)
+
+        assert math.isnan(res.value)
+        assert res.loc.line == 8
+        assert res.loc.column == 2
+        assert res.loc.stream_name == "float"
+
+    @pytest.mark.parametrize(
+        "value", [
+            "abc", "1.2.3", "NaN", "INF", ".NAn", " ", "   ", "\n", "\t",
+            "1 2", "https://url", ":bare", "multi\n  line",
+        ],
+    )
+    def test_construct_str(self, value):
+        mark = Mark(None, None, 0, 0, None, None)
+        node = ScalarNode(None, value, start_mark=mark)
+        res = Constructor("float").construct_yaml_str(node)
+
+        assert res.value == value
+        assert res.loc.line == 1
+        assert res.loc.column == 1
+        assert res.loc.stream_name == "float"
+
+    def test_construct_seq(self):
+        mark = Mark(None, None, 4, 3, None, None)
+        children = [
+            ScalarNode("tag:yaml.org,2002:int", str(i), start_mark=mark)
+            for i in range(4)
+        ]
+        node = SequenceNode(None, children, start_mark=mark)
+        # Sequences are single-element generators (lazy parsing) that we need
+        # to exhaust. Simplest way of doing this is to assing to a single
```

Comment:
I'd google the word "assing" to see if it means what it means, but it sounds NSFW.